Don't create a new Drupal account when site settings don't allow it

florisla opened the pull request:
Currently, d8-google-login creates a new account in Drupal if the used Google account does not match any known account.

This is also the case if the site is configured to *not* accept users to self-register.

This change fixes that -- the account is only created if setting user.settings.register is set to
'visitors'.

Commit:
Don't create a new account when site settings don't allow it

Commit:
Improve code comment a bit

Commit:
Document the self-registration site setting in README

nathandentzau commented on src/Controller/GoogleOAuthController.php:
```diff
@@ -69,6 +69,17 @@ public function authenticate() {
     $user = user_load_by_mail($user_email);
 
     if (!$user) {
+      $user_config = \Drupal::config('user.settings');
```
Please inject the config object using the `ContainerInjectionInterface`: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21DependencyInjection%21ContainerInjectionInterface.php/interface/ContainerInjectionInterface/8.2.x

nathandentzau commented on src/Controller/GoogleOAuthController.php:
```diff
@@ -69,6 +69,17 @@ public function authenticate() {
     if (!$user) {
+      $user_config = \Drupal::config('user.settings');
+      $registration_allowed = $user_config->get('register');
+      if ('visitors' != $registration_allowed) {
```
I'm not a huge fan of yoda conditions, but since we're comparing strings we should use type comparison `!==`.

nathandentzau commented:
Hi @florisla thank you for your PR. I've pointed out a couple items. If you could please address them we can roll this in. Thanks!

Commit:
Use proper !== operator for string comparison

Also, inverse the order of the 'if' condition -- because
Yoda order preferred is not.

Commit:
Inject configuration through ContainerInjectionInterface

florisla commented:
@nathandentzau This seems to work.

Not sure about the naming and styling though (`configFactory` vs `config_factory`) --  I took inspiration from here:
http://cgit.drupalcode.org/fotorama_gallery/tree/src/Plugin/Field/FieldFormatter/FotoramaGalleryFormatter.php.

Update: fixed.

Commit:
Use consistent variable casing (snake_case)

Consistency is advised at
https://www.drupal.org/docs/develop/standards/coding-standards#naming .

nathandentzau commented:
Hi @florisla thank you for taking the time to contribute to our project. This looks good to me. Have a great day!

/approved
